### DISCUSSION TITLE
Setup require("neoconf").setup() should be run BEFORE setting up any lsp server with lspconfig

### DISCUSSION THREAD

**Discussion by jaredmcqueen** [LazyVim/LazyVim#481] [Q&A] [answered]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

NVIM v0.8.3

### Operating system/version

MacOS 13.2.1

### Describe the bug

I'm trying to disable LSP warnings that my yaml keys are in the wrong order.  I'm using `yamlls` and it is attached to the buffer.  The warnings look like this and are unsightly with large yaml files:
<img width="855" alt="image" src="https://user-images.githubusercontent.com/3491953/227098230-d74ae286-0b97-4cdb-858c-9fa3c14e7384.png">

In order to disable these orderings I think I need to add a custom configuration.  Here's what I'm trying:
```lua
return {
  require("lspconfig").yamlls.setup({
    settings = {
      yaml = {
        keyOrdering = false,
      },
    },
  }),
}
```

When I go to open up neovim, I get the following error and I still see the annoying ordering warnings.

<img width="1184" alt="image" src="https://user-images.githubusercontent.com/3491953/227097825-82667940-092c-457b-a49f-46a7cc922105.png">

Lastly, I can confirm that `yamlls` is attached to my buffer:
<img width="785" alt="image" src="https://user-images.githubusercontent.com/3491953/227098080-a1cd962a-80fe-4fd8-a66b-e2c0f7e42ca5.png">

After searching tons of issues and looking at example configurations, I can't seem to figure out how to suppress these key ordering warnings.

### Steps To Reproduce

1. Install `yaml-language-server` using Mason
2. Open a yaml file with the wrong ordering of keys
```yaml
a:
  true
z:
  true
b:
  false
```
4. See the warning.
5. Attempt a user-defined configuration called `yaml.lua`:

```lua
return {
  require("lspconfig").yamlls.setup({
    settings = {
      yaml = {
        keyOrdering = false,
      },
    },
  }),
}
```

See the error:
<img width="614" alt="image" src="https://user-images.githubusercontent.com/3491953/227099059-e32b6392-0bad-4984-88f1-b4e362194a6d.png">

### Expected Behavior

warnings are suppressed

### Repro

```Lua
-- DO NOT change the paths and don't remove the colorscheme
local root = vim.fn.fnamemodify("./.repro", ":p")

-- set stdpaths to use .repro
for _, name in ipairs({ "config", "data", "state", "cache" }) do
  vim.env[("XDG_%s_HOME"):format(name:upper())] = root .. "/" .. name
end

-- bootstrap lazy
local lazypath = root .. "/plugins/lazy.nvim"
if not vim.loop.fs_stat(lazypath) then
  vim.fn.system({ "git", "clone", "--filter=blob:none", "https://github.com/folke/lazy.nvim.git", lazypath, })
end
vim.opt.runtimepath:prepend(lazypath)

-- install plugins
local plugins = {
  "folke/tokyonight.nvim",
  "folke/LazyVim",
  -- add any other plugins here
}
require("lazy").setup(plugins, {
  root = root .. "/plugins",
})

vim.cmd.colorscheme("tokyonight")
-- add anything else here

return {
  require("lspconfig").yamlls.setup({
    settings = {
      yaml = {
        keyOrdering = false,
      },
    },
  }),
}
```

**Comment by folke** [comment]
Please use discussions for user questions next time and please read the docs / examples
LSP servers are setup by lazyvim https://www.lazyvim.org/plugins/lsp#nvim-lspconfig

This is how you configure lsp servers. I'm sure you can figure it out for yaml:

```lua
return {
{
    "neovim/nvim-lspconfig",
    ---@class PluginLspOpts
    opts = {
      ---@type lspconfig.options
      servers = {
        -- pyright will be automatically installed with mason and loaded with lspconfig
        pyright = {},
      },
    },
  },
}
```

**Comment by jaredmcqueen** [comment]
ok, i'll jam on this and post a solution when I get it.  Thanks for a nudge in the right direction

**Comment by jaredmcqueen** [comment, marked as answer]
answer can be found here:
https://github.com/folke/dot/blob/master/nvim/lua/plugins/lsp.lua#L90-L96

```lua
return {
  {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        yamlls = {
          settings = {
            yaml = {
              keyOrdering = false,
            },
          },
        },
      },
    },
  },
}
```

**Comment by kalleh** [comment]
As a newb, I fell into the same trap as the OP. And while I did read the documentation snippet pointed to, the confusion stems from the documentation of the lsp itself saying something completely different (in this case gleam).   Something like, "some lsp server may tell you to configure the lsp with `require("lspconfig").XXXX.setup`, please use the following method instead.
